web/src/main/java/com/marklogic/hub/web/model/FlowStepModel.java

```diff
@@ -0,0 +1,182 @@
+        public String targetEntity;
+    }
+
+    @JsonProperty("steps")
+    private List<StepSummary> stepModels = new ArrayList<>();
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    public void setId(String id) {
+        this.id = id;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public void setDescription(String description) {
+        this.description = description;
+    }
+
+    public int getBatchSize() {
+        return batchSize;
+    }
+
+    public void setBatchSize(int batchSize) {
+        this.batchSize = batchSize;
+    }
+
+    public int getThreadCount() {
+        return threadCount;
+    }
+
+    public void setThreadCount(int threadCount) {
+        this.threadCount = threadCount;
+    }
+
+    public boolean isStopOnError() {
+        return stopOnError;
+    }
+
+    public void setStopOnError(boolean stopOnError) {
+        this.stopOnError = stopOnError;
+    }
+
+    public JsonNode getOptions() {
+        return options;
+    }
+
+    public void setOptions(JsonNode options) {
+        this.options = options;
+    }
+
+    public List<StepSummary> getStepModels() {
+        return stepModels;
+    }
+
+    public void setStepModels(List<StepSummary> stepModels) {
+        this.stepModels = stepModels;
+    }
+
+    public static Map<String, Step> createFlowSteps(Flow flow, JSONObject jsonObject) {
+        flow.setDescription(jsonObject.getString("description"));
+        flow.setId(jsonObject.getString("id", flow.getId() != null ? flow.getId() : UUID.randomUUID().toString()));
+        flow.setBatchSize(jsonObject.getInt("batchSize", FlowImpl.DEFAULT_BATCH_SIZE));
+        flow.setThreadCount(jsonObject.getInt("threadCount", FlowImpl.DEFAULT_THREAD_COUNT));
+        flow.setOptions(jsonObject.getNode("options"));
+        flow.setStopOnError(jsonObject.getBoolean("stopOnError", FlowImpl.DEFAULT_STOP_ONERROR));
+        flow.setVersion(jsonObject.getInt("version", 1));
+
+        Map<String, Step> stepMaps = new LinkedHashMap<>();
+        if (!jsonObject.isExist("steps")) {
+            return stepMaps;
+        }
+        List<Object> steps = jsonObject.getArray("steps");
+        Map<String, Step> existingSteps = flow.getSteps();
+
+        Map<String, String> keyById = new HashMap<>();
+        for (String key : existingSteps.keySet()) {
+            keyById.put(existingSteps.get(key).getId(), key);
+        }
+        final String[] count = {"1"};
+        steps.forEach(s -> {
+            JSONObject stepJson = new JSONObject((JsonNode) s);
+            String id = stepJson.getString("id");
+            String stepName = stepJson.getString("name");
+            Step step;
+            if (keyById.containsKey(id)) { //update
+                step = existingSteps.get(keyById.get(id));
+            } else {
```

Comment:
Not required as Request is not going to update Step in here.
Any update to the Steps will go via the Steps API
Only reordering of Steps will be done via Flows endpoint.